packages/server/test/unit/routes_spec.ts

```diff
@@ -0,0 +1,230 @@
+import type { NetworkProxy } from '@packages/proxy'
+import type HttpProxy from 'http-proxy'
+import type { RemoteStates } from '../../lib/remote_states'
+
+import chai, { expect } from 'chai'
+import sinon from 'sinon'
+import proxyquire from 'proxyquire'
+import { Cfg } from '../../lib/project-base'
+
+chai.use(require('@cypress/sinon-chai'))
+
+describe('lib/routes', () => {
+  // https://github.com/cypress-io/cypress/issues/25891
+  describe('https upgrade fix', () => {
+    let routeOptions
+
+    beforeEach(() => {
+      sinon.restore()
+
+      routeOptions = {
+        config: {
+          clientRoute: 'clientRoute',
+          namespace: 'namespace',
+        } as Cfg,
+        getSpec: sinon.stub().returns({}),
+        // @ts-ignore
+        networkProxy: {
+          handleHttpRequest: () => {},
+        } as NetworkProxy,
+        nodeProxy: {} as HttpProxy,
+        onError: () => {},
+        // @ts-ignore
```

Comment:
```suggestion
        // @ts-expect-error
```

Reply:
Addressed in [`8f9cd45`](https://github.com/cypress-io/cypress/pull/28650/commits/8f9cd45dd94c928569fd6401764e7719208b1940)